libs/isograph-react/src/core/cache.ts

```diff
@@ -410,7 +444,10 @@ function normalizeDataIntoRecord(
     }
   }
   if (recordHasBeenUpdated) {
-    mutableEncounteredIds.add(targetParentRecordId);
+    mutableEncounteredIds[targetParentRecordId.concreteType] ??= new Set();
```

Comment:
likewise, let's assign this to an intermediate value (`const foo = mutableEncounteredIds[targetParentRecordId.concreteType] ??= new Set(); foo.add(...)`